src/components/CharacterTab.jsx

```diff
@@ -222,6 +232,33 @@ export default function CharacterTab() {
     cellStyle: { display: 'flex' },
   }), [])
 
+  const externalFilterChanged = useCallback((newValue) => {
+    setCurrentFilters(newValue)
+    gridRef.current.api.onFilterChanged()
+  }, [])
+
+  const charinfo = JSON.parse(JSON.stringify(DB.getMetadata().characters))
+
+  const doesExternalFilterPass = useCallback(
+    (node) => {
+      if (currentFilters.element.length && !currentFilters.element.includes(charinfo[node.data.id].element)) {
```

Comment:
A small readability improvement here, instead of charinfo[node.data.id] everywhere we can name the variable as filteredCharacter

```
    const characterMetadata = Utils.clone(DB.getMetadata().characters)
    const filteredCharacter = characterMetadata[node.data.id]

    ... currentFilters.element.includes(filteredCharacter.element) ...
```